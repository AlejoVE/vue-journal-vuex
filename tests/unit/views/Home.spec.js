import { shallowMount} from '@vue/test-utils'
import Home from '@/views/Home.vue'

describe('Tests in Home View', () => {
    test('Should match snapshot', () => {
        const wrapper = shallowMount(Home);
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Click in button should redirect to no-entry', () => {
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
    })
})
