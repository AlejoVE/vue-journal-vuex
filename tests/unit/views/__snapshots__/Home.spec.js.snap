// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Home View Should match snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Home</h1><button class="btn btn-primary">primary</button><button class="btn btn-secondary">secondary</button><button class="btn btn-success">success</button>
</div>
`;
